### PR TITLE
Meta: Add script to enforce license headers & run it on Travis

### PR DESCRIPTION
**Pull request by linusg**
Depends on https://github.com/SerenityOS/serenity/pull/2169 being merged first!

### CODE REVIEW

File: Meta/check-license-headers.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    fi
+done < <(git ls-files -- \
+'*.cpp' \
+'*.h' \
+':!:Tests' \
+':!:Base' \
+':!:Kernel/FileSystem/ext2_fs.h' \
+':!:Libraries/LibC/getopt.cpp' \
+':!:Libraries/LibCore/puff.h' \
+':!:Libraries/LibELF/exec_elf.h' \
```
Comment (linusg) [inline Meta/check-license-headers.sh:27]: These are borrowed from somewhere and while they do contain a license header it doesn't match ours.

If someone adds a borrowed file where this is the case as well, it has to be added here manually.

File: Meta/check-license-headers.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# We simply check if the file starts with:
+# /*
+#  * Copyright
+PATTERN=$'^/\*\n \* Copyright'
+ERRORS=()
+
+while IFS= read -r f; do
+    if [[ ! -s $f ]]; then
+        continue
+    fi
```
Comment (linusg) [inline Meta/check-license-headers.sh:15]: Apparently we have empty `.h` files - skip those.

Comment (awesomekling) [reply Meta/check-license-headers.sh:15]: Uhh, we can also remove them :D

File: Meta/check-license-headers.sh (hunk 1 above)
Comment (awesomekling) [reply Meta/check-license-headers.sh:27]: Yes.. although we should strive to get rid of these and try to avoid adding new ones :)

**Comment by linusg** [comment]
image

:+1:

**Comment by linusg** [comment]
let me rebase this real quick and it should be good to go!

**Pull request merged**